### PR TITLE
Refactor WMFLocationManager to Swift

### PR DESCRIPTION
**Pull request by VojtaStavik**
Hey Wikipedia iOS team,

This PR is a byproduct of our case study about a real-world Objective-C to Swift refactoring.

### What's happening in this PR:
- Unused code is removed from `WMFMockLocationManager`.
- The original ObjC version of `WMFMockLocationManager` is made testable.
- Tests are added to cover its functionality.
- A new Swift implementation is written based on the test suite.
- Objective-C compatibility is added using a standalone protocol and (in theory) you should be able to just revert the `Add ObjC support for LocationManager` commit when you no longer need it.
- The implementations are swapped and the original version is deleted. 

We had to reverse engineer some of the original functionality and it'd be great to hear your feedback.

---

**Note:** Since the PR is quite big, we tried to keep the commits small and structured the history such that you can easily review it commit by commit.

### CODE REVIEW

**Comment by tonisevener** [comment]
@VojtaStavik apologies for the radio silence - this is great, thank you so much for doing this! This came in during the middle of a large release cycle for us, so we haven't had a chance to look deeply yet. We're wrapping that up now and will be taking a look at this soon.

File: WMF Framework/LocationManager.swift [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    /// Set the delegate if you want to receive location and heading updates.
+    public var delegate: LocationManagerDelegate?
+
+    /// Returns the current locationManager permission state.
+    public var autorizationStatus: CLAuthorizationStatus { type(of: locationManager).authorizationStatus() }
+
+    /// Starts monitoring location and heading updates.
+    public func startMonitoringLocation() {
+        guard autorizationStatus != .notDetermined else {
+            authorize(succcess: startMonitoringLocation)
```
Comment (tonisevener) [inline WMF Framework/LocationManager.swift:58]: Minor spelling - should be `success`.

File: Wikipedia/Code/ArticleLocationCollectionViewController.swift [hunk 2]
```diff
@@ -7,7 +7,7 @@ class ArticleLocationCollectionViewController: ColumnarCollectionViewController,
         }
     }
     let dataStore: MWKDataStore
-    fileprivate let locationManager = WMFLocationManager.fine()
+    fileprivate let locationManager = LocationManager()
```
Comment (tonisevener) [inline Wikipedia/Code/ArticleLocationCollectionViewController.swift:10]: I wonder if we ought to reference `LocationManagerProtocol` as much as possible now that you have the abstraction. `PlacesViewController` could be updated to use it too.

**Comment by tonisevener** [comment]
@VojtaStavik We are ready to merge this in. Will you be able to update this with the latest from `master` (we have since changed our default branch to that from `develop`) and fix the merge conflict? If not, we will try to update it from our side later this week. Thanks!

**Comment by VojtaStavik** [comment]
Hi @tonisevener, thank you for your patience and for the comments on the PR. Unfortunately, the situation has changed a little bit, and I'm not able to finish the PR. Feel free to update it yourself and finish it :)

**Comment by joewalsh** [comment]
Closing in favor of #3613 which is an updated version of these changes. Thanks!

**Pull request closed without merging**